### PR TITLE
Shift zip generation to pup.

### PR DESCRIPTION
**Pull request by Camwyn**
Restructuring the PRs for the Wizard for cleaner CR.
START HERE!

**REQUIRES/USES**: https://github.com/the-events-calendar/tribe-common/pull/2292

### Notes on tests:
The deprecation test is going to fail currently - ignore it as I'm mocking a major version for this until we get the correct version. As TBDs will cause `pup build` to fail.

The test we **_need_** to pass its the zip generation. There is an issue with the build of said zip internally (the zip may not work) that has to do with the React code and something that has changed, but not yet in the bucket. See [this line](https://github.com/the-events-calendar/the-events-calendar/actions/runs/12014363007/job/33489973535?pr=4838#step:16:467) if interested, it is OK to ignore for now - a lot of that code has changed drastically in upcoming PRs.

### 🎫 Ticket

[TEC-5329]

### 🗒️ Description

- Add pup via phar.
- Adjust installed deps (composer and npm) for a clean build
- Correct StellarWP translations URL.
Hat-tip to @jesseeproductions for doing a lot of this work on another plugin that served as an example.

### ✔️ Checklist
- [ ] Ran `npm run changelog` to add changelog file(s). More info [here](https://docs.theeventscalendar.com/developer/git/changelogs/#process)
- [ ] Code is covered by **NEW** `wpunit` or `integration` tests.
- [x] Code is covered by **EXISTING** `wpunit` or `integration` tests.
- [x] Are all the **required** tests passing?
- [x] Automated code review comments are addressed.
- [ ] Have you added Artifacts? No: a working zip is the artifact - see zip test
- [x] Check the base branch for your PR.
- [x] Add your PR to the project board for the release.

[TEC-5329]: https://stellarwp.atlassian.net/browse/TEC-5329?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -121,9 +113,9 @@
     "mt-a11y-dialog": "^5.1.1",
     "prop-types": "^15.6.2",
     "querystringify": "^2.0.0",
-    "react": "^18.0.0",
+    "react": "^16.14.0",
```
Comment (jesseeproductions) [inline package.json:116]: There has been issues with the latest version of WP. Are we updating to React 18 or keeping it on the old version

Comment (jesseeproductions) [reply package.json:116]: Same for WordPress/Scripts. Should it be included?

File: .github/workflows/zip.yml [hunk 2]
```diff
@@ -1,37 +1,50 @@
...
     runs-on: ubuntu-latest
     steps:
-      - name: If a pull_request, set final to 0
-        if: github.event_name == 'pull_request'
-        run: |
-          echo "JOB_FINAL=0" >> $GITHUB_ENV
-          echo "JOB_BRANCH=${{ github.event.pull_request.head.ref }}" >> $GITHUB_ENV
-      - name: If a workflow_dispatch, set final based on input
-        if: github.event_name == 'workflow_dispatch'
+      - uses: actions/checkout@v3
```
Comment (jesseeproductions) [inline .github/workflows/zip.yml:31]: Can this one use v4?

File: package.json (hunk 1 above)
Comment (Camwyn) [reply package.json:116]: I've been holding off on the update until the blocks issues are resolved.
I expect that before this goes out...
I don't want to accidentally create more issues 😉

File: .github/workflows/zip.yml (hunk 2 above)
Comment (Camwyn) [reply .github/workflows/zip.yml:31]: Oh, for sure!

File: package.json (hunk 1 above)
Comment (Camwyn) [reply package.json:116]: As far as `@wordpress/scripts` - we are aiming to stop including them (any WordPress libraries) manually and just depend on the existing ones from the WP install.

**Pull request merged**